src/akamai/js/akamaiHD_flash.js

```diff
@@ -201,8 +211,29 @@ require("../../../html5-common/js/utils/constants.js");
      * @returns {boolean} True or false indicating success
      */
     this.setVideoUrl = function(url, encoding) {
-      /* Should be set to false if stream URL changes */
-      loaded = false;
+      newController=this.controller;
+      if (currentUrl.replace(/[\?&]_=[^&]+$/,'') != url)
+      {
+        currentUrl = url || "";
+
+        // bust the chrome caching bug
+        if (currentUrl.length > 0) 
+        {
+          currentUrl = currentUrl + (/\?/.test(currentUrl) ? "&" : "?") + "_=" + getRandomString();
+        }
+        urlChanged = true;
+        loaded = false;
+        url = "setVideoUrl("+currentUrl+")";
+      }
+      if (_.isEmpty(currentUrl)) 
+      {
+        this.controller.notify(this.controller.EVENTS.ERROR, { errorcode: 0 }); 
```

Comment:
For ads we do wish to support empty stream url.  Is it possible to not throw an error here?